File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> mp ;
        mp[0]=1;
        int s=0,count=0;
        for(int i:nums)
        {
            s+=i;
            if(mp.find(s-k)!=mp.end())
            {
                count+=mp[s-k];
            }
            if(mp.find(s)!=mp.end())
            {
                mp[s]+=1;
            }
            else
            {
                mp[s] =1;
            }
        }
        return count;
    }
};

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int s=0,count =0;
        HashMap<Integer,Integer> mp = new HashMap<>();
        mp.put(0,1);
        for(int i:nums)
        {
            s+=i;
            if(mp.containsKey(s-k))
            {
                count+=mp.get(s-k);
            }
            if(mp.containsKey(s))
            {
                mp.put(s,mp.get(s)+1);
            }
            else
            {
                mp.put(s,1);
            }
        }
        return count;
    }
}